querydsl-core/src/main/java/com/mysema/commons/lang/Pair.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021, The Querydsl Team (http://www.querydsl.com/team)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mysema.commons.lang;
```

Comment:
I doubt mysema.com appreciates us shipping new code under their name.

However, as Pair is in the API and SPI, moving this to `com.querydsl` will be a breaking API change and force us to release a 6.0.0 version. Which is also very likely to break existing third party plugins. So not sure what to do here.

Reply:
I think this is the best we can do without a breaking release. I would be okay with going for 6.0.0, but in that case I would favor following springs lead and dropping java<17 support as well.

Reply:
I just feel like this project is already moving very  slowly, supporting less versions could help.

Reflecting on this PR again, I have to correct myself it has nothing to do with java version support. 

Just let me know if you want to go the breaking way then I'll adjust it.

Reply:
Lets move these classes to the `com.querydsl` package name while we can. 5.x has been out for a while and I don't mind making the next release a 6.0.